### PR TITLE
Need A Way To Pass In File Path For Override Cred Provider

### PR DESCRIPTION
**Pull request by sethAmazon**
**Description:** 
Bug fix
Need a way to pass in string to creds provider 

**Link to tracking Issue:** 
N/A

**Testing:** 
N/A

**Documentation:** 
N/A

### CODE REVIEW

File: internal/aws/awsutil/conn.go [hunk 1]
```diff
@@ -183,14 +183,27 @@ func GetAWSConfigSession(logger *zap.Logger, cn ConnAttr, cfg *AWSSessionSetting
...
-	credProviders = append(credProviders, overrideCredProviders...)
-	config.Credentials = credentials.NewCredentials(&credentials.ChainProvider{
-		Providers: credProviders,
-	})
+	// do not overwrite for sts assume role
+	if cfg.RoleARN == "" {
+		defaultCredProviders := defaults.CredProviders(config, defaults.Handlers())
+		overrideCredProviders := override.GetCredentialsChainOverride().GetCredentialsChain()
+		credProviders := make([]credentials.Provider, 0)
+		if cfg.SharedCredentialsFile != nil {
```
Comment (sethAmazon) [inline internal/aws/awsutil/conn.go:191]: Need this for when not using default cred file and profile. Ex container insights.

File: override/aws/credential.go [hunk 2]
```diff
@@ -27,15 +27,15 @@ var credentialsChainOverride *CredentialsChainOverride
 
 func GetCredentialsChainOverride() *CredentialsChainOverride {
 	if credentialsChainOverride == nil {
-		credentialsChainOverride = &CredentialsChainOverride{credentialsProvider: make([]credentials.Provider, 0)}
+		credentialsChainOverride = &CredentialsChainOverride{credentialsProvider: make([]func(string) credentials.Provider, 0)}
```
Comment (sethAmazon) [inline override/aws/credential.go:30]: Need to be able to pass in string because the override is a location.

File: internal/aws/awsutil/conn.go [hunk 3]
```diff
@@ -183,15 +183,36 @@ func GetAWSConfigSession(logger *zap.Logger, cn ConnAttr, cfg *AWSSessionSetting
...
+}
+
+func customCredentialProvider(cfg *AWSSessionSettings, config *aws.Config) []credentials.Provider {
 	defaultCredProviders := defaults.CredProviders(config, defaults.Handlers())
 	overrideCredProviders := override.GetCredentialsChainOverride().GetCredentialsChain()
-	credProviders := make([]credentials.Provider, 0, len(defaultCredProviders)+len(overrideCredProviders))
+	credProviders := make([]credentials.Provider, 0)
+	// if is for differently configured shared creds file location
+	// else if is for diff profile but no change in creds file ex run in containers
+	if cfg.SharedCredentialsFile != nil && len(cfg.SharedCredentialsFile) > 0 {
```
Comment (sethAmazon) [inline internal/aws/awsutil/conn.go:202]: Did not consider when we change the file or profile location.

File: internal/aws/awsutil/conn.go [hunk 4]
```diff
@@ -183,15 +183,36 @@ func GetAWSConfigSession(logger *zap.Logger, cn ConnAttr, cfg *AWSSessionSetting
...
-	credProviders = append(credProviders, overrideCredProviders...)
-	config.Credentials = credentials.NewCredentials(&credentials.ChainProvider{
-		Providers: credProviders,
-	})
-	return config, s, nil
+	for _, provider := range overrideCredProviders {
+		for _, file := range cfg.SharedCredentialsFile {
+			credProviders = append(credProviders, provider(file))
+		}
+	}
```
Comment (SaxyPandaBear) [inline internal/aws/awsutil/conn.go:214]: These are ordered, right? Does it matter where these override credentials are in the list? Like, if the override is the last thing in the credential provider list, will it break out and use something before we get to this?

**Review by SaxyPandaBear — APPROVED** [review]

File: internal/aws/awsutil/conn.go (hunk 4 above)
Comment (sethAmazon) [reply internal/aws/awsutil/conn.go:214]: I will respond in slack since we can't talk here bc it is internal.

**Review by SaxyPandaBear — APPROVED** [review]

**Review by zhihonl — APPROVED** [review]

**Pull request merged**